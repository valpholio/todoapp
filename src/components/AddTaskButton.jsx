import React from 'react';
import PropTypes from 'prop-types';

const AddTaskButton = ({
                           enterAddTaskMode = () => {}
                       }) => {
    return (
        <div onClick={enterAddTaskMode}>
            + Add Task
        </div>
    );
};

AddTaskButton.propTypes = {
    enterAddTaskMode: PropTypes.func
};

export default AddTaskButton;