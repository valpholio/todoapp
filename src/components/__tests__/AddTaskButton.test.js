import React from 'react';
import renderer from 'react-test-renderer';
import AddTaskButton from '../AddTaskButton.jsx';

describe('AddTaskButton', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<AddTaskButton />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});