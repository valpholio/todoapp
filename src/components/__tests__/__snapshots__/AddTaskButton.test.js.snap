// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTaskButton snapshot renders 1`] = `
<button
  className="btn btn-default btn-sm"
  onClick={[Function]}
>
  <span
    className="glyphicon glyphicon-plus"
  />
   Add Task
</button>
`;
